packages/playground/__tests__/context/messages/reducer.test.tsx

```diff
@@ -0,0 +1,89 @@
+import { reducer, defaultState, Action } from '~/context/messages/reducer';
```

Comment:
Generally, please use the Given, When, Then pattern for tests:

// Given
... here test setup
// When
... here test interaction
// Then
.. her test result verification

e.g.:

```
test('When message of type "DONE" is dispatched', () => {
      // Given
      const status = 'DONE';
      const content = 'System message content';
      const action: Action = {
        type: 'LOG_SYSTEM',
        payload: {
          content: content,
          status: status,
        },
      };

     // When
      const state = reducer(defaultState, action);

      // Then 
      expect(state.messages[0].id).toBe(0);
      expect(state.messages[0].prompt).toBe('SYSTEM');
      expect(state.messages[0].status).toBe(status);
      expect(state.messages[0].content).toBe(content);
      expect(state.messages[0].severity).toBe('Success');
      expect(state.nextId).toEqual(0);
    });
```